x/auth/legacy/v0_36/types.go

```diff
@@ -0,0 +1,19 @@
+// DONTCOVER
+// nolint
+package v0_36
+
+import v034auth "github.com/cosmos/cosmos-sdk/x/auth/legacy/v0_34"
+
+const (
+	ModuleName = "auth"
+)
+
+type (
+	GenesisState struct {
+		Params v034auth.Params `json:"params"`
```

Comment:
We shouldn't refer to old version types in the new version, only the migrate function should know about them, if we make the types cross-dependant across versions, in few version will be hard to handle

Reply:
basically, the same comment everywhere a 034 type is imported into a 036 type, without polluting with comments :) 
It should instead be defined locally and the migrate function should take care of converting

Reply:
Those types are already defined on the 0.34 version. What you are suggesting leads to a lot of code duplication

Reply:
That was the initial goal of the legacy types, to act as independent snapshots.

Reply:
If it's defined in the previous version there's no need to define it again, just import it.

Reply:
This is version-specific code. I don't see this as code duplication. Perhaps this whole thing should not live within the SDK/design should be rediscussed.

IMHO this version-specific collection of data types shoud live without depend on each other.